src/AudioTrackManager.ts

```diff
@@ -0,0 +1,93 @@
+import { AudioTrack } from "./types";
+import MediaAudio from "./MediaAudio";
+
+class AudioTrackManager {
+  public isPlaying: boolean;
+
+  private currentAudio: MediaAudio[];
+
+  constructor() {
+    this.isPlaying = false;
+    this.currentAudio = [];
+  }
+
+  public play(audioTracks: AudioTrack[]): void {
+    if (audioTracks.length === 0) return;
+
+    this.isPlaying = true;
+
+    audioTracks.forEach((audioTrack) => {
+      if (!audioTrack.url) return;
+
+      // Get the index of this current audio track if it exists.
+      const index = this.currentAudio.findIndex(
+        (currentAudio) => currentAudio.url === audioTrack.url,
+      );
+
+      // If the track already exists in currentAudio
+      if (index === -1) {
+        const audio = new MediaAudio(audioTrack.url);
+
+        audio.loop = audioTrack.loop;
+        audio.volume = audioTrack.volume;
+        audio.originalVolume = audio.volume;
+        audio.fastSeek(0);
+        audio.play();
+        audio.onended = () => {
+          this.currentAudio = this.currentAudio.filter(
+            (currentAudio) => currentAudio !== audio,
+          );
+        };
+
+        this.currentAudio.push(audio);
+      } else {
+        // Check for tracks that need to stop playing.
+        if (audioTrack.stopPlaying) {
+          this.currentAudio[index].pause();
+          this.currentAudio = this.currentAudio.filter(
+            (currentAudio) => currentAudio.url !== audioTrack.url,
+          );
+          return;
+        }
+
+        // Check if any tracks need to be restarted.
+        if (audioTrack.behaviour === "restart") {
+          if (index !== -1) {
```

Comment:
This is guaranteed because if `index === -1` we would not be in this else block from line 28 and 43